proposals/security_and_privacy.md

```diff
@@ -0,0 +1,277 @@
+good security, and manage the data that is
+collected appropriately.
+
+The metadata provided in the WoT Thing Description
+can be helpful in this regard since the security mechanisms
+supported by a device can be easily inspected and,
+via semantic annotations, the type of data provided
+by the device can be determined.
+
+### Does this specification allow an origin access to aspects of a user’s local computing environment?
+
+No.  Actually, there is no executable content
+included in data or metadata provided by WoT Things.
+The only operations that a WoT Server will perform
+on behalf of a WoT Client are those that it chooses
+to support via its exposed interactions (network API).
+
+### Does this specification allow an origin access to other devices? 
+
+Yes, as discussed above.  A WoT Thing can act as a gateway to 
+other WoT Things or to other non-IP protocols.
+
+The situation is unavoidably more complex than the client-server
+architecture of broswers.  A better analogy than browsers
+would be the microservice architecture used by many web services.
+The WoT Architecture supports best practices there, such
+as Zero-Trust networks.
+
+### Does this specification allow an origin some measure of control over a user agent’s native UI? 
+
+No.  The WoT Architecture is M2M and makes no mention of direct
+user interfaces.
+
+That said, some devices connected via the WoT may have
+local user interfaces which might be exposed to control
+over network interactions.  Exposure of such controls
+is at the discretion of individual device manufacturers,
+who should do an analysis of security risks as part of
+their software development process.
+
+### Does this specification expose temporary identifiers to the web?
+
+The WoT Thing Description has a required element (not actually
+mentioned in the current document, the WoT Architecture, but
+in the WoT Thing Description) which is meant to be a unique ID
+(a URN, actually).  This is necessary to support Linked Data.
+In fact, some domains, such as medical devices in the US,
+have a legal requirement to support immutable identifiers.
+
+However, we suggest that the tracking risk this poses shold
+be mitigated, whenever possible (eg if not legally disallowed
+as discussed above) by allowing such identifiers to be modified
+at least when the devices are reprovisioned.
+
+Generally, though, the identifiers in Thing Descriptions 
+are relatively long-lived.  
+
+### Does this specification distinguish between behavior in first-party and third-party contexts?
+
+This concept is not applicable to the WoT context.
+
+### How should this specification work in the context of a user agent’s "incognito" mode?
+
+This concept is not applicable to the WoT context, as there is no user agent.
+
+### Does this specification persist data to a user’s local device? 
+
+Potentially, as a WoT Thing may include interactions that
+support POST requests.  IoT devices can act like web servers that 
```

Comment:
The question is what is meant with "the user's local device". I would interpret it as the device, from which the user accesses and controls an IoT device. 
The specification does not persist data, it is completely under control of the user's device (and also of the IoT device) whether it persists some state changes.

Reply:
Later on I talk about this more.   In the "normal" web, the device is a client, but downloaded scripts can persist state, which causes various issues (for example, tracking).   I was thinking here about the case of a personally owned device which is a server, which some other device posts data to and then retrieves it later.   There is nothing stopping someone from creating a WoT device that acts like a blackboard (for example, to allow messages to be posted to a watch display) and allows set and get of strings, for instance.   Such a WoT Thing would be privacy risk since a tracking ID could be posted to it...

Reply:
OK, I updated this to talk about both cases: the device is a server, and the device is a client.  Persistent store-and-return data storage is possible in both cases.  Mitigation in both cases is to use access controls and establish trust before storing and returning data.